### ISSUE TITLE
Add support for hostnames in the settings

### ISSUE THREAD

**Issue by meehien** [Sonarr/Sonarr#1335]
Would it be possible to remove the "valid IP formatting" check for the "bind address" field in settings (settings -> general -> bind address) .

I like to use hostames to address network devices in my LAN and I also run a webserver on the same device that also runs sonarrr. In order for my setup to work I have to fill the bind address with the local hostname.

This setup works great, the problem is that currently the way to do it is to manually edit the config.xml file as the web interface complains about the not valid IP. Also, if any changes need to be made afterwards using the web interface, the interface will complain about this field until is changed back to an IP formatted string. I think removing the check is a trivial change to do, won't affect many users, and it would provide a lot flexibility for users like me (even more so because sonarr does work if that field does not contain an IP).

**Comment by markus101** [comment]
That field is quite a mess and doesn't work the way it really should, the intention is to bind to a specific IP address, but in reality it binds to a specific URL, which is why its working in your case. The fact that its binding to a specific URL is a bug and will be fixed once we have a web server that works properly with it (likely in v3). With the intention being to fix the bug I don't think adding further support for the buginess of it makes sense.

**Comment by meehien** [comment]
Sure, but you have just now arrived at v2. Also this is just string validation and presumably if you do get that web server in v3 you will probably want to support hostnames for address binding as an alternative to ip addresses (e.g. localhost instead of 127.0.0.1).

**Comment by markus101** [comment]
> Sure, but you have just now arrived at v2

v2 has been out for a long time now, more than 2 years I believe.

>  if you do get that web server in v3 you will probably want to support hostnames for address binding as an alternative to ip addresses (e.g. localhost instead of 127.0.0.1).

The intention there is the listen on one IP interface only, so hostname would be irrelevant, this is useful for multi-IP systems or to force Sonarr to only listen on the loopback address so only the local system can access it (useful for security or to use a reverse proxy in front of it). The issue with the current system is it alters the HTTP.sys bindings (in Windows) so only the IP or hostname enter works, which allows you to have multiple things listening on the same port, but different hostnames as HTTP.sys handles the connections for you (which is how full blown IIS works), under mono things work differently and setting a bind address can do some weird things and lock you out of the UI. HTTP.sys doesn't fit well with Sonarr because it requires an administrator to make changes otherwise you're restricted to localhost access only. Overall the biggest issues are administrative requirements and inconsistencies, but also SSL certificates are a big pain.

**Comment by meehien** [comment]
> The intention there is the listen on one IP interface only, so hostname would be irrelevant, this is useful for multi-IP systems or to force Sonarr to only listen on the loopback address so only the local system can access it.

This is exactly what I am using it for. Maybe I didn't express myself very well. Let me try again: 
probably because of the inconsistencies you specified, if `bind address` is an ip address then sonarr internal web server will not listen to a `http://hostname:port`, it will only listen to `http://x.x.x.x:port`.

As such I found the workaround to bind a hostname (which mono internally presumably resolves it to the interface/ip it's assigned to) so that sorarr will reply when I type `http://hostname:port` in my browser. This is on par with what regular web servers (e.g. nginx, lighttpd, apache) with support for virtual hosts do: you need to listen on an interface for a specific "virtual host".

I guess I'd like to better ask: do you plan on supporting Sonarr listening to URLs of type 
`http://hostname:port` on multi-IP systems, perhaps by having the `bind address` field to specify the interface and an additional field where you can input the hostname you want?

**Comment by Taloth** [comment]
meehien, there are some complications with supporting such vhosts across all OS, so we like to keep binding to ip only.
But you could set up an nginx proxy, which would have the added benefit of being able to access all your different services on one host:port.
